### PR TITLE
Changing Hugo to Gotham 

### PR DESCRIPTION
**Pull request by mibragimchayev**
fixes: #56

### CODE REVIEW

File: tpl/tplimpl/embedded/templates/_default/rss.xml [hunk 1]
```diff
@@ -16,7 +16,7 @@
     <title>{{ if eq  .Title  .Site.Title }}{{ .Site.Title }}{{ else }}{{ with .Title }}{{.}} on {{ end }}{{ .Site.Title }}{{ end }}</title>
     <link>{{ .Permalink }}</link>
     <description>Recent content {{ if ne  .Title  .Site.Title }}{{ with .Title }}in {{.}} {{ end }}{{ end }}on {{ .Site.Title }}</description>
-    <generator>Hugo -- gohugo.io</generator>{{ with .Site.LanguageCode }}
+    <generator>Gotham -- gothamhq.com</generator>{{ with .Site.LanguageCode }}
```
Comment (felicianotech) [inline tpl/tplimpl/embedded/templates/_default/rss.xml:19]: This is not relevant to the meta generator tag and the bug we're fixing. This isn't even HTML. I suggest we put this piece into its own PR.

Also, the file above this, you manually changed one and the other was generated right? Did you do that on purpose?

File: common/hugo/hugo_test.go [hunk 2]
```diff
@@ -30,7 +30,7 @@ func TestHugoInfo(t *testing.T) {
 	c.Assert(hugoInfo.CommitHash, qt.Equals, commitHash)
 	c.Assert(hugoInfo.BuildDate, qt.Equals, buildDate)
 	c.Assert(hugoInfo.Environment, qt.Equals, "production")
-	c.Assert(string(hugoInfo.Generator()), qt.Contains, fmt.Sprintf("Hugo %s", hugoInfo.Version()))
+	c.Assert(string(hugoInfo.Generator()), qt.Contains, fmt.Sprintf("Gotham %s", hugoInfo.Version()))
```
Comment (felicianotech) [inline common/hugo/hugo_test.go:33]: Does this pass?

Comment (mibragimchayev) [reply common/hugo/hugo_test.go:33]: No this does not pass because hugoInfo is based on CurrentVersion.Version()

File: tpl/tplimpl/embedded/templates/_default/rss.xml (hunk 1 above)
Comment (mibragimchayev) [reply tpl/tplimpl/embedded/templates/_default/rss.xml:19]: Yes, I will pull this out of this PR and do it separately.

No, I changed both. Will need to review both of these files in the separate PR.

Comment (felicianotech) [reply tpl/tplimpl/embedded/templates/_default/rss.xml:19]: Gotcha. So we don't want to do that here. Files in the tpl path that end with `autogen.go` are automatically generated from the other tpl files. So never edit those directly.

If you edit something in tpl such as `tpl/tplimpl/embedded/templates/_default/rss.xml`, after you save the file (and any other tpl changes), you can run `mage generate` to run the Go generation script which will create the other files. Then commit those changes.

This process isn't documented as far as I know and I had to learn the hardway. I created an Issue for us to document this going forward: https://github.com/gothamhq/gotham/issues/58

File: common/hugo/hugo_test.go (hunk 2 above)
Comment (felicianotech) [reply common/hugo/hugo_test.go:33]: Instead of `hugoInfo.Version()` we'd use `GothamVersion.String()`.

Also, a couples up there's a related test that's checking the type. That test we can remove. Our version string is just a string, not a custom type.
